apps/api-extractor-model/src/model/ApiEnum.ts

```diff
@@ -71,4 +72,11 @@ export class ApiEnum extends ApiItemContainerMixin(ApiNameMixin(ApiReleaseTagMix
     }
     super.addMember(member);
   }
+
+  /** @beta @override */
+  public buildCanonicalReference(): DeclarationReference {
+    return (this.parent ? this.parent.canonicalReference : DeclarationReference.empty())
+      .addNavigationStep(Navigation.Exports, this._getCanonicalReferenceName())
```

Comment:
Will you only have `ApiEnum` nodes that are guaranteed to be exported, or do you anticipate non-exported enums? This is perfectly legal TypeScript:

```ts
// foo.d.ts
const enum NotExported { Foo, Bar }
export function f(): NotExported;
```

The return type of `f` should have the reference `foo!~NotExported:enum`, but this indicates the enum will always be an export and not a local (i.e. non-export).

Reply:
Today API Extractor considers `NotExported` to be essentially nameless (e.g. its name is free to be mangled in the .d.ts rollup).  We don't expect there to be a documentation page for it.  The [ae-forgotten-export](https://api-extractor.com/pages/messages/ae-forgotten-export/) warning is reported, and currently it does not appear in the .api.json file at all.

The philosophy is like:

_"If the type is so important that you'd make a website page for it, then shouldn't it have a name so we can talk about it?  Shouldn't that name be exported so that developers can defer to it in a type annotation?"_

This is subjective, however.  If people can provide compelling justifications, we may need to support it in the future.